### ISSUE TITLE
Teuchos: Build warning in BigUInt

### ISSUE THREAD

**Issue by mhoemmen** [trilinos/Trilinos#2316]
```
In file included from .../Trilinos/packages/teuchos/core/src/Teuchos_PrintDouble.cpp:43:
.../Trilinos/packages/teuchos/core/src/Teuchos_BigUInt.hpp:180:9: warning: variable 'xi' is used uninitialized whenever 'if' condition is false [-Wsometimes-uninitialized]
    if (i + ndigits < n) xi = x[i + ndigits] >> nbits;
        ^~~~~~~~~~~~~~~
.../Trilinos/packages/teuchos/core/src/Teuchos_BigUInt.hpp:254:10: note: in instantiation of member function 'Teuchos::BigUInt<34>::operator>>=' requested here
  factor >>= 1;
         ^
.../Trilinos/packages/teuchos/core/src/Teuchos_PrintDouble.cpp:193:27: note: in instantiation of function template specialization 'Teuchos::operator/<34>' requested here
      auto d_np1 = r_x_10 / s;
                          ^
.../Trilinos/packages/teuchos/core/src/Teuchos_BigUInt.hpp:182:12: note: uninitialized use occurs here
    x[i] = xi;
           ^~
.../Trilinos/packages/teuchos/core/src/Teuchos_BigUInt.hpp:180:5: note: remove the 'if' if its condition is always true
    if (i + ndigits < n) xi = x[i + ndigits] >> nbits;
    ^~~~~~~~~~~~~~~~~~~~~
.../Trilinos/packages/teuchos/core/src/Teuchos_BigUInt.hpp:179:21: note: initialize the variable 'xi' to silence this warning
    std::uint32_t xi;
                    ^
                     = 0
```

@trilinos/teuchos

**Comment by ibaned** [comment]
Thanks for pointing this out, @mhoemmen ! It looks like a legitimate uninitialized variable bug. I created  pull request #2321 to fix it.

**Comment by mhoemmen** [comment]
Thanks @ibaned !  I would have fixed myself but wanted to make sure the algorithm was right.  I just merged your PR (#2321).
